kpatch-build/create-diff-object.c

```diff
@@ -294,17 +295,32 @@ static int rela_equal(struct rela *rela1, struct rela *rela2)
 		if (!toc_relasec1)
 			ERROR("cannot find .rela.toc");
 
+		insn1[0] = insn1[1] = 0x0;
 		r_toc_relasec1 = find_rela_by_offset(toc_relasec1, rela1->addend);
-		if (!r_toc_relasec1)
-			ERROR(".toc entry not found %s + %x", rela1->sym->name, rela1->addend);
+		if (!r_toc_relasec1) {
+			if (!find_toc_const(toc_relasec1->base, &insn1[0], rela1->addend))
```

Comment:
Thanks for the review.
> why is the list of toc_consts needed?

Consider the example of `sched/fair.o` described at `find_toc_const`. Arithmetic in a function may add constant entries to `.toc section`, constants here are instructions to store/load value to/from registers.  `.toc section` is populated with relocation entries only for most of the object files but its one of the  `PowerPC` quirks, which needs to be handled. These constants/instructions are not part of `.rela.toc section` but are part of  `.toc section` only.

> Here it should be straightforward to just read the .toc section directly, since the original rela already points to it.

Good idea, they can be read from `.toc section + offset` because they are not relocation entries, so not part of `.rela.toc section`. Which will be far clean and simple than the approached taken by me.